.changeset/heavy-onions-buy.md

```
@@ -0,0 +1,49 @@
+  model: {
+    id: 'custom-provider/model',
+    url: 'https://api.custom.com/v1',
+    apiKey: 'key',
+    headers: {
+      'X-Custom-Header': 'custom-value',
+    },
+  },
+});
+```
+
+Fixes https://github.com/mastra-ai/mastra/issues/9760
```

markdownlint-cli2 (0.18.1)

Bare URL used

(MD034, no-bare-urls)